pkg/cnb/mountable_buildpack_layer_test.go

```diff
@@ -0,0 +1,165 @@
+package cnb
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/random"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/sclevine/spec"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pivotal/kpack/pkg/apis/experimental/v1alpha1"
+)
+
+func TestMountableBuildpackLayer(t *testing.T) {
+	spec.Run(t, "Test Mountable Buildpack Layer", testMountableBuildpackLayer)
+}
+
+func testMountableBuildpackLayer(t *testing.T, when spec.G, it spec.S) {
+	handler := http.NewServeMux()
+	server := httptest.NewServer(handler)
+	tagName := fmt.Sprintf("%s/some/image:tag", server.URL[7:])
+
+	var registryCalls []string
+	var backingRandomLayer v1.Layer
+	var layer v1.Layer
+
+	it.Before(func() {
+		var err error
+
+		backingRandomLayer, err = random.Layer(1000, types.DockerLayer)
+		require.NoError(t, err)
+
+		handler.HandleFunc("/v2/", func(writer http.ResponseWriter, request *http.Request) {
+			registryCalls = append(registryCalls, request.RequestURI)
+
+			writer.WriteHeader(200)
+		})
+
+		digest, err := backingRandomLayer.Digest()
```

Comment:
just an aesthetic preference but can be move the `handler.HandleFunc()` together and the assignment of digest, diffID and size together?

Reply:
👍